### PR TITLE
Kyma CLI inline documentation review

### PR DESCRIPTION
**Pull request by bszwarc**
**Description**

Changes proposed in this pull request:

- Review all inline documentation consistency - initial draft for install, provision, and uninstall commands. 
- to do: internal directory (errors, log messages) 

**Related issue(s)**
See also #39

### CODE REVIEW

**Comment by bszwarc** [comment]
> Looks good. Merge it. The only thing I'm not sure about if log messages should be capitalized. Usually, log messages are printed in lower case, since it is easier to find them later in the log storage. @a-thaler?
Thanks @igor-karpukhin  :) I will have a second look at it - if they shouldn't be capitalised, I will change this :) I also improved the README.md file to make it compatible with our [template](https://github.com/kyma-project/community/blob/master/guidelines/internal-guidelines/repository-template/template/README.md)

**Comment by bszwarc** [comment]
I also have a question @a-thaler , @lilitgh - can we have a WIP label added? It's quite useful :D

File: README.md [hunk 1]
```diff
@@ -2,77 +2,103 @@
...
+
+For the installation instructions, see the [release page](https://github.com/kyma-project/cli/releases).
+
+## Usage
+
+### Available commands
+
+Kyma CLI comes with a set of commands:
 
 - `version`: Shows the Kyma cluster version and the Kyma CLI version
```
[the remark this answers is not in the document]
Comment (kazydek) [reply README.md]: Here and everywhere below?

File: README.md [hunk 2]
```diff
@@ -2,77 +2,103 @@
...
-- `provision minikube`: Initializes minikube with a new cluster (replaces the `minikube.sh` script) 
+- `provision minikube`: Initializes Minikube with a new cluster (replaces the `minikube.sh` script)
```
Comment (kazydek) [inline README.md]:
```suggestion
- `provision minikube` initializes Minikube on a new cluster. It replaces the `minikube.sh` script. 
```

File: README.md [hunk 3]
```diff
@@ -2,77 +2,103 @@
...
+## Usage
+
+### Available commands
+
+Kyma CLI comes with a set of commands:
 
 - `version`: Shows the Kyma cluster version and the Kyma CLI version
-- `provision minikube`: Initializes minikube with a new cluster (replaces the `minikube.sh` script) 
+- `provision minikube`: Initializes Minikube with a new cluster (replaces the `minikube.sh` script) 
 - `install`: Installs Kyma to a cluster based on a release (replaces the `ìnstaller.sh` and `is-installed.sh` script)
```
Comment (kazydek) [inline README.md]:
```suggestion
- `install` installs Kyma on a cluster based on a release. It replaces `ìnstaller.sh` and `is-installed.sh` scripts.
```

Comment (kazydek) [reply README.md]: Based on the **current** release?

File: README.md [hunk 4]
```diff
@@ -2,77 +2,103 @@
...
 - `uninstall`: Uninstalls all Kyma related resources from a cluster
```
Comment (kazydek) [inline README.md]:
```suggestion
- `uninstall` uninstalls all Kyma-related resources from a cluster.
```

File: README.md [hunk 5]
```diff
@@ -2,77 +2,103 @@
...
-- `completion`: Outputs shell completion code for bash
+- `completion`: Generates and shows the bash completion script
```
Comment (kazydek) [inline README.md]:
```suggestion
- `completion` generates and shows the bash completion script.
```

File: README.md [hunk 6]
```diff
@@ -2,77 +2,103 @@
...
 - `test`: Triggers and reports the tests for every Kyma module
```
Comment (kazydek) [inline README.md]:
```suggestion
- `test` triggers and reports tests for every Kyma module.
```

File: README.md [hunk 7]
```diff
@@ -2,77 +2,103 @@
...
-- `help`: Displays usage for the given command (e.g. `kyma help`, `kyma help status`, etc...)
-
-## Prerequisites
+- `help`: Displays and explains usage of a given command
```
Comment (kazydek) [inline README.md]:
```suggestion
- `help` displays and explains the usage of a given command.
```

File: README.md [hunk 8]
```diff
@@ -2,77 +2,103 @@
...
-```bash
-kyma provision minikube
-kyma install
 ```
+kyma {COMMAND} {FLAGS}
+```
+where:
 
-Installation of Kyma with minikube on Windows:
+* **{COMMAND}** specifies the operation you want to perform
```
Comment (kazydek) [inline README.md]:
```suggestion
* **{COMMAND}** specifies the operation you want to perform.
```

File: README.md [hunk 9]
```diff
@@ -2,77 +2,103 @@
...
+* **{FLAGS}** specify optional flags. For example, `-v` or `--verbose` for additional information on performed operations.
```
Comment (kazydek) [inline README.md]:
```suggestion
* **{FLAGS}** specify optional flags. For example, use `-v` or `--verbose` for additional information on performed operations.
```

File: README.md [hunk 10]
```diff
@@ -2,77 +2,103 @@
...
+## Development
 
-## Kyma CLI as a Kubectl plugin
+### Kyma CLI as a kubectl plugin
 
-To follow this section a kubectl version of 1.12.0 or later is required.
+> **NOTE**: To use Kyma CLI as a kubectl plugin, use Kubernetes version 1.12.0 or higher.
 
-A plugin is nothing more than a standalone executable file, whose name begins with kubectl- . To install a plugin, simply move this executable file to anywhere on your PATH.
+A plugin is a standalone executable file with a name prefixed with `kubectl-` . To use the plugin perform the following steps:
```
Comment (kazydek) [inline README.md]:
```suggestion
A plugin is a standalone executable file with a name prefixed with `kubectl-` . To use the plugin, perform the following steps:
```

File: README.md [hunk 11]
```diff
@@ -2,77 +2,103 @@
...
 
-Rename a `kyma` binary to `kubectl-kyma` and place it anywhere in your PATH:
+1. Rename a `kyma` binary to `kubectl-kyma` and place it anywhere in your **{PATH}**:
```
Comment (kazydek) [inline README.md]:
```suggestion
1. Rename the `kyma` binary to `kubectl-kyma` and place it anywhere under your **{PATH}**:
```

File: README.md [hunk 12]
```diff
@@ -2,77 +2,103 @@
...
 
 ```bash
 sudo mv ./kyma /usr/local/bin/kubectl-kyma
 ```
 
-Run `kubectl plugin list` command and you will see your plugin in the list of available plugins.
+2. Run `kubectl plugin list` command to see your plugin on the list of available plugins.
```
Comment (kazydek) [inline README.md]:
```suggestion
2. Run the `kubectl plugin list` command to see your plugin on the list of all available plugins.
```

File: README.md [hunk 13]
```diff
@@ -2,77 +2,103 @@
...
+3. Invoke your plugin as a kubectl command:
 
 ```bash
 $ kubectl kyma version
 Kyma CLI version: v0.6.1
 Kyma cluster version: 1.0.0
 ```
 
-To know more about extending kubectl with plugins read [kubernetes documentation](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/).
+For more information on extending kubectl with plugins read [Kubernetes documentation](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/).
```
Comment (kazydek) [inline README.md]:
```suggestion
For more information on extending kubectl with plugins, read [Kubernetes documentation](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/).
```

File: internal/kubectl/kubectl.go [hunk 14]
```diff
@@ -177,7 +177,7 @@ func IsPodReady(namespace string, labelName string, labelValue string, verbose b
 		if containerStatus != "true" {
 			events, err := RunCmd(verbose, "get", "event", "-n", namespace, "-o", "go-template='{{range .items}}{{if eq .involvedObject.name \"'"+pod+"'\"}}{{.message}}{{\"\\n\"}}{{end}}{{end}}'")
 			if err != nil {
-				fmt.Printf("Error while checking for pod events '%s'\n‚", err)
+				fmt.Printf("Error occurred while checking for Pod Events '%s'\n‚", err)
```
Comment (kazydek) [inline internal/kubectl/kubectl.go]:
```suggestion
				fmt.Printf("Error occurred while checking Pod Events '%s'\n‚", err)
```

File: pkg/kyma/cmd/install/cmd.go [hunk 15]
```diff
@@ -64,31 +64,31 @@ func NewCmd(o *Options) *cobra.Command {
...
 	cobraCmd.Flags().BoolVarP(&o.Local, "local", "l", false, "Install from sources")
-	cobraCmd.Flags().StringVarP(&o.LocalSrcPath, "src-path", "", "", "Path to local sources to use")
-	cobraCmd.Flags().StringVarP(&o.LocalInstallerVersion, "installer-version", "", "", "Version of installer docker image to use while building locally")
-	cobraCmd.Flags().StringVarP(&o.LocalInstallerDir, "installer-dir", "", "", "Directory of installer docker image to use while building locally")
-	cobraCmd.Flags().DurationVarP(&o.Timeout, "timeout", "", 0, "Timeout after which CLI should give up watching installation")
-	cobraCmd.Flags().StringVarP(&o.Password, "password", "p", "", "Pre-defined cluster password")
-	cobraCmd.Flags().VarP(&o.OverrideConfigs, "override", "o", "Path to YAML file with parameters to override. Multiple entries of this flag allowed")
+	cobraCmd.Flags().StringVarP(&o.LocalSrcPath, "src-path", "", "", "Path to local sources")
+	cobraCmd.Flags().StringVarP(&o.LocalInstallerVersion, "installer-version", "", "", "The version of Kyma Installer docker image used for local installation")
```
Comment (kazydek) [inline pkg/kyma/cmd/install/cmd.go]:
```suggestion
	cobraCmd.Flags().StringVarP(&o.LocalInstallerVersion, "installer-version", "", "", "Version of the Kyma Installer Docker image used for local installation")
```

File: pkg/kyma/cmd/install/cmd.go [hunk 16]
```diff
@@ -64,31 +64,31 @@ func NewCmd(o *Options) *cobra.Command {
...
+	cobraCmd.Flags().StringVarP(&o.LocalInstallerDir, "installer-dir", "", "", "The directory of Kyma Installer docker image used for local installation")
```
Comment (kazydek) [inline pkg/kyma/cmd/install/cmd.go]:
```suggestion
	cobraCmd.Flags().StringVarP(&o.LocalInstallerDir, "installer-dir", "", "", "Directory of the Kyma Installer Docker image used for local installation")
```

File: pkg/kyma/cmd/install/cmd.go [hunk 17]
```diff
@@ -64,31 +64,31 @@ func NewCmd(o *Options) *cobra.Command {
...
+	cobraCmd.Flags().DurationVarP(&o.Timeout, "timeout", "", 0, "The timeout after which CLI stops watching the installation progress")
+	cobraCmd.Flags().StringVarP(&o.Password, "password", "p", "", "The predefined cluster password")
+	cobraCmd.Flags().VarP(&o.OverrideConfigs, "override", "o", "The path to YAML file with parameters to override. Multiple entries of this flag are allowed")
```
Comment (kazydek) [inline pkg/kyma/cmd/install/cmd.go]:
```suggestion
	cobraCmd.Flags().VarP(&o.OverrideConfigs, "override", "o", "Path to the YAML file with parameters to override. Multiple entries of this flag are allowed")
```

File: pkg/kyma/cmd/install/cmd.go [hunk 18]
```diff
@@ -178,15 +178,15 @@ func (cmd *command) validateFlags() error {
 		if cmd.opts.LocalSrcPath == "" {
 			goPath := os.Getenv("GOPATH")
 			if goPath == "" {
-				return fmt.Errorf("No local 'src-path' configured and no applicable default found, verify if you have exported a GOPATH?")
+				return fmt.Errorf("No local 'src-path' configured and no applicable default found, have you exported a GOPATH?")
```
Comment (kazydek) [inline pkg/kyma/cmd/install/cmd.go]:
```suggestion
				return fmt.Errorf("No local 'src-path' configured and no applicable default found. Check if you exported a GOPATH.")
```

File: pkg/kyma/cmd/install/cmd.go [hunk 19]
```diff
@@ -178,15 +178,15 @@ func (cmd *command) validateFlags() error {
...
 			}
 			cmd.opts.LocalSrcPath = filepath.Join(goPath, "src", "github.com", "kyma-project", "kyma")
 		}
 		if _, err := os.Stat(cmd.opts.LocalSrcPath); err != nil {
-			return fmt.Errorf("Configured 'src-path=%s' does not exist, please check if you configured a valid path", cmd.opts.LocalSrcPath)
+			return fmt.Errorf("Configured 'src-path=%s' does not exist, check if you configured a valid path", cmd.opts.LocalSrcPath)
```
Comment (kazydek) [inline pkg/kyma/cmd/install/cmd.go]:
```suggestion
			return fmt.Errorf("Configured 'src-path=%s' does not exist. Check if you configured a valid path.", cmd.opts.LocalSrcPath)
```

File: pkg/kyma/cmd/install/cmd.go [hunk 20]
```diff
@@ -178,15 +178,15 @@ func (cmd *command) validateFlags() error {
...
 		}
 		if _, err := os.Stat(filepath.Join(cmd.opts.LocalSrcPath, "installation", "resources")); err != nil {
-			return fmt.Errorf("Configured 'src-path=%s' seems to not point to a Kyma repository, please verify if your repository contains a folder 'installation/resources'", cmd.opts.LocalSrcPath)
+			return fmt.Errorf("Configured 'src-path=%s' seems to not point to a Kyma repository, verify if your repository contains the 'installation/resources' folder", cmd.opts.LocalSrcPath)
```
Comment (kazydek) [inline pkg/kyma/cmd/install/cmd.go]:
```suggestion
			return fmt.Errorf("Configured 'src-path=%s' does not seem to point to a Kyma repository. Check if your repository contains the 'installation/resources' folder", cmd.opts.LocalSrcPath)
```

File: pkg/kyma/cmd/install/cmd.go [hunk 21]
```diff
@@ -250,7 +250,7 @@ func (cmd *command) configureHelm() error {
 
 	data, ok := cfg["data"].(map[interface{}]interface{})
 	if !ok {
-		return fmt.Errorf("unable to get data from helm secret")
+		return fmt.Errorf("Unable to get data from Helm secret")
```
Comment (kazydek) [inline pkg/kyma/cmd/install/cmd.go]:
```suggestion
		return fmt.Errorf("Unable to get data from the Helm Secret")
```

File: pkg/kyma/cmd/install/cmd.go [hunk 22]
```diff
@@ -274,7 +274,7 @@ func (cmd *command) configureHelm() error {
 func writeHelmFile(data map[interface{}]interface{}, helmData string, helmHome string, filename string) error {
 	value, ok := data[helmData].(string)
 	if !ok {
-		return fmt.Errorf("unable to get %s from helm secret data", helmData)
+		return fmt.Errorf("Unable to get %s from Helm secret data", helmData)
```
Comment (kazydek) [inline pkg/kyma/cmd/install/cmd.go]:
```suggestion
		return fmt.Errorf("Unable to get %s from Helm Secret data", helmData)
```

File: pkg/kyma/cmd/install/cmd.go [hunk 23]
```diff
@@ -64,31 +64,31 @@ func NewCmd(o *Options) *cobra.Command {
...
-	cobraCmd.Flags().StringVarP(&o.LocalInstallerDir, "installer-dir", "", "", "Directory of installer docker image to use while building locally")
-	cobraCmd.Flags().DurationVarP(&o.Timeout, "timeout", "", 0, "Timeout after which CLI should give up watching installation")
-	cobraCmd.Flags().StringVarP(&o.Password, "password", "p", "", "Pre-defined cluster password")
-	cobraCmd.Flags().VarP(&o.OverrideConfigs, "override", "o", "Path to YAML file with parameters to override. Multiple entries of this flag allowed")
+	cobraCmd.Flags().StringVarP(&o.LocalSrcPath, "src-path", "", "", "Path to local sources")
+	cobraCmd.Flags().StringVarP(&o.LocalInstallerVersion, "installer-version", "", "", "The version of Kyma Installer docker image used for local installation")
+	cobraCmd.Flags().StringVarP(&o.LocalInstallerDir, "installer-dir", "", "", "The directory of Kyma Installer docker image used for local installation")
+	cobraCmd.Flags().DurationVarP(&o.Timeout, "timeout", "", 0, "The timeout after which CLI stops watching the installation progress")
+	cobraCmd.Flags().StringVarP(&o.Password, "password", "p", "", "The predefined cluster password")
```
Comment (kazydek) [inline pkg/kyma/cmd/install/cmd.go]:
```suggestion
	cobraCmd.Flags().StringVarP(&o.Password, "password", "p", "", "Predefined cluster password")
```

File: pkg/kyma/cmd/install/cmd.go [hunk 24]
```diff
@@ -64,31 +64,31 @@ func NewCmd(o *Options) *cobra.Command {
...
-	cobraCmd.Flags().StringVarP(&o.LocalInstallerDir, "installer-dir", "", "", "Directory of installer docker image to use while building locally")
-	cobraCmd.Flags().DurationVarP(&o.Timeout, "timeout", "", 0, "Timeout after which CLI should give up watching installation")
-	cobraCmd.Flags().StringVarP(&o.Password, "password", "p", "", "Pre-defined cluster password")
-	cobraCmd.Flags().VarP(&o.OverrideConfigs, "override", "o", "Path to YAML file with parameters to override. Multiple entries of this flag allowed")
+	cobraCmd.Flags().StringVarP(&o.LocalSrcPath, "src-path", "", "", "Path to local sources")
+	cobraCmd.Flags().StringVarP(&o.LocalInstallerVersion, "installer-version", "", "", "The version of Kyma Installer docker image used for local installation")
+	cobraCmd.Flags().StringVarP(&o.LocalInstallerDir, "installer-dir", "", "", "The directory of Kyma Installer docker image used for local installation")
+	cobraCmd.Flags().DurationVarP(&o.Timeout, "timeout", "", 0, "The timeout after which CLI stops watching the installation progress")
```
Comment (kazydek) [inline pkg/kyma/cmd/install/cmd.go]:
```suggestion
	cobraCmd.Flags().DurationVarP(&o.Timeout, "timeout", "", 0, "Timeout after which CLI stops watching the installation progress")
```

File: pkg/kyma/cmd/uninstall/cmd.go [hunk 25]
```diff
@@ -40,20 +40,20 @@ func NewCmd(o *Options) *cobra.Command {
...
-- Removes tiller
-- Removes the Kyma installer
+		Short: "Uninstalls Kyma from a running Kubernetes cluster",
+		Long: `Uninstalls Kyma from a running Kubernetes cluster.
+
+Make sure that your KUBECONFIG is already pointing to the target cluster.
+This command:
+- Removes your cluster administrator account
+- Removes Tiller
+- Removes the Kyma Installer
```
Comment (kazydek) [inline pkg/kyma/cmd/uninstall/cmd.go]:
```suggestion
- Removes Kyma Installer
```

File: README.md [hunk 26]
```diff
@@ -2,77 +2,103 @@
...
+    ```bash
+    kyma provision minikube
+    kyma install
+    ```
+
+* Install Kyma with Minikube on Windows:
+
+    ```bash
+    kyma provision minikube
+    # follow instructions to add hosts
```
Comment (kazydek) [inline README.md]: what does that refer to?

File: pkg/kyma/cmd/uninstall/cmd.go [hunk 27]
```diff
@@ -306,8 +306,8 @@ func (cmd *command) waitForInstallerToUninstall() error {
 			case "Error":
 				if !errorOccured {
 					errorOccured = true
-					cmd.CurrentStep.Failuref("Error uninstalling Kyma: %s", desc)
-					cmd.CurrentStep.LogInfof("To fetch the logs from the installer execute: 'kubectl logs -n kyma-installer -l name=kyma-installer'")
+					cmd.CurrentStep.Failuref("Kyma uninstallation failed: %s", desc)
```
Comment (kazydek) [inline pkg/kyma/cmd/uninstall/cmd.go]:
```suggestion
					cmd.CurrentStep.Failuref("Failed to uninstall Kyma: %s", desc)
```

File: README.md [hunk 28]
```diff
@@ -2,77 +2,103 @@
...
 
-A command line tool to support developers of and with [Kyma](https://github.com/kyma-project/kyma)
+Kyma CLI is a command line tool which supports [Kyma](https://github.com/kyma-project/kyma) developers. It provides a set of commands you can use to install and test Kyma. 
 
-## Available Commands
+## Prerequisites
+
+Kyma CLI requires the following software:
+- [kubectl](https://github.com/kubernetes/kubectl) 
+- [minikube](https://github.com/kubernetes/minikube)
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply README.md]: Done.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (bszwarc) [reply README.md]: Done.

File: README.md (hunk 4 above)
Comment (bszwarc) [reply README.md]: Done.

File: README.md (hunk 8 above)
Comment (bszwarc) [reply README.md]: Done.

File: README.md (hunk 9 above)
Comment (bszwarc) [reply README.md]: Done.

File: README.md (hunk 10 above)
Comment (bszwarc) [reply README.md]: Done.

File: README.md (hunk 12 above)
Comment (bszwarc) [reply README.md]: Done

File: internal/kubectl/kubectl.go (hunk 14 above)
Comment (bszwarc) [reply internal/kubectl/kubectl.go]: I would rephrase it to `searching for` to keep the meaning.

File: pkg/kyma/cmd/install/cmd.go (hunk 15 above)
Comment (bszwarc) [reply pkg/kyma/cmd/install/cmd.go]: Done.

File: pkg/kyma/cmd/install/cmd.go (hunk 16 above)
Comment (bszwarc) [reply pkg/kyma/cmd/install/cmd.go]: Done.

File: pkg/kyma/cmd/install/cmd.go (hunk 24 above)
Comment (bszwarc) [reply pkg/kyma/cmd/install/cmd.go]: Done.

File: pkg/kyma/cmd/install/cmd.go (hunk 23 above)
Comment (bszwarc) [reply pkg/kyma/cmd/install/cmd.go]: Done.

File: pkg/kyma/cmd/install/cmd.go (hunk 17 above)
Comment (bszwarc) [reply pkg/kyma/cmd/install/cmd.go]: Done.

File: pkg/kyma/cmd/install/cmd.go (hunk 18 above)
Comment (bszwarc) [reply pkg/kyma/cmd/install/cmd.go]: Done.

File: pkg/kyma/cmd/install/cmd.go (hunk 19 above)
Comment (bszwarc) [reply pkg/kyma/cmd/install/cmd.go]: Done.

File: pkg/kyma/cmd/install/cmd.go (hunk 20 above)
Comment (bszwarc) [reply pkg/kyma/cmd/install/cmd.go]: Done.

File: pkg/kyma/cmd/install/cmd.go (hunk 21 above)
Comment (bszwarc) [reply pkg/kyma/cmd/install/cmd.go]: Done.

File: pkg/kyma/cmd/install/cmd.go (hunk 22 above)
Comment (bszwarc) [reply pkg/kyma/cmd/install/cmd.go]: Done.

File: pkg/kyma/cmd/uninstall/cmd.go (hunk 25 above)
Comment (bszwarc) [reply pkg/kyma/cmd/uninstall/cmd.go]: Done.

File: pkg/kyma/cmd/uninstall/cmd.go (hunk 27 above)
Comment (bszwarc) [reply pkg/kyma/cmd/uninstall/cmd.go]: Done.
